### PR TITLE
MOD-7299: Exact TAG using double quotes

### PR DESCRIPTION
**Pull request by nafraf**
**Describe the changes in the pull request**

1. Support double quotes for exact Tag, while not breaking exact Text.
2. Support numeric operators without fixing PR #4604 since this PR introduces breaking changes (requires adding more specific rules to the lexer)
3. Remove parser/v3, DIALECT 5

**Which issues this PR fixes**
1. MOD-7299

**Main objects this PR modified**
1. ...
2. ...

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: tests/pytests/test_tags.py [hunk 1]
```diff
@@ -416,15 +417,15 @@ def testTagAutoescaping(env):
     env.cmd('HSET', '{doc}:22', 'tag', "w'?*1'", 'id', '22')
 
     # Test exact match
-    res = env.cmd('FT.SEARCH', 'idx', '@tag:{abc:1}', 'NOCONTENT',
+    res = env.cmd('FT.SEARCH', 'idx', '@tag:{"abc:1"}', 'NOCONTENT',
```
Comment (MeirShpilraien) [inline tests/pytests/test_tags.py:420]: Do we have a test that `|` works inside the`{}`, for example: `@tag:{"foo"|"bar"}`

Comment (nafraf) [reply tests/pytests/test_tags.py:423]: Test included in lines 481-510.

File: src/query_parser/v2/lexer.rl [hunk 2]
```diff
@@ -25,6 +25,67 @@ void RSQuery_Parse_v2(void *yyp, int yymajor, QueryToken yyminor, QueryParseCtx
 void *RSQuery_ParseAlloc_v2(void *(*mallocProc)(size_t));
 void RSQuery_ParseFree_v2(void *p, void (*freeProc)(void *));
 
+int RSQuery_ParseNumericOp_v2(void* pParser, int OperatorType, QueryToken tok, 
+      QueryParseCtx *q, const char *ts, const char *te, char c1,
+      unsigned int opLen) {
+    tok.len = te - (ts + 1);
+    int len = tok.len;
```
Comment (oshadmi) [inline src/query_parser/v2/lexer.rl:32]: unused variable?

File: src/query_parser/v2/lexer.rl [hunk 3]
```diff
@@ -25,6 +25,67 @@ void RSQuery_Parse_v2(void *yyp, int yymajor, QueryToken yyminor, QueryParseCtx
...
+    tok.pos = ts - q->raw;
+    tok.s = ts + 1;
+
+    // Find the position before the operator
+    char *end1 = strchr(tok.s, c1) - 1;
+    // Find the position after the operator
+    char *start2 = end1 + opLen + 1;
+    // Remove spaces before the operator
+    while (isspace(*end1)) {
```
Comment (oshadmi) [inline src/query_parser/v2/lexer.rl:41]: Can a modifier have escaped spaces? e.g.,
"@mod\\\\ >42"

File: src/query_parser/v2/lexer.rl [hunk 4]
```diff
@@ -296,6 +430,21 @@ main := |*
       fbreak;
     }
   };
+
+  suffix_exact => {
+    tok.type = QT_TERM;
+    tok.len = te - (ts + 3); // remove the quotes and the star at the end
```
Comment (oshadmi) [inline src/query_parser/v2/lexer.rl:436]:
```suggestion
    tok.len = te - (ts + 3); // remove the quotes at the end
```

File: src/query_parser/v2/lexer.rl [hunk 5]
```diff
@@ -311,6 +460,20 @@ main := |*
     }
   };
 
+  contains_exact => {
+    tok.type = QT_TERM;
+    tok.len = te - (ts + 4); // remove the quotes and the stars
```
Comment (oshadmi) [inline src/query_parser/v2/lexer.rl:465]:
```suggestion
    tok.len = te - (ts + 4); // remove the quotes and the star
```

File: src/query_parser/v2/lexer.rl (hunk 2 above)
Comment (nafraf) [reply src/query_parser/v2/lexer.rl]: Yes. `len` variable was removed. Thanks

File: src/query_parser/v2/lexer.rl (hunk 4 above)
Comment (nafraf) [reply src/query_parser/v2/lexer.rl]: Done.

File: src/query_parser/v2/lexer.rl (hunk 5 above)
Comment (nafraf) [reply src/query_parser/v2/lexer.rl]: Fixed.

File: src/query_parser/v2/lexer.rl [hunk 6]
```diff
@@ -105,6 +115,111 @@ main := |*
...
+
+  mod_not_equal => {
+    printf("mod_not_equal\n");
+    tok.pos = ts-q->raw;
+    tok.len = te - (ts + 1) - 2;
+    tok.s = ts+1;
+
+    // remove unescaped trailing spaces
+    while(tok.len > 1 && isspace(tok.s[tok.len - 1]) 
+            && tok.s[tok.len - 2] != '\\') {
```
[the remark this answers is not in the document]
Comment (nafraf) [reply src/query_parser/v2/lexer.rl]: Yes, it was wrong, Fixed by [afb43ad](https://github.com/RediSearch/RediSearch/pull/4772/commits/afb43ad06a44c535a602164d58f4bbdeac7be9ae)

File: src/query_parser/v2/lexer.rl (hunk 3 above)
Comment (nafraf) [reply src/query_parser/v2/lexer.rl]: Yes. Supported by [afb43ad](https://github.com/RediSearch/RediSearch/pull/4772/commits/afb43ad06a44c535a602164d58f4bbdeac7be9ae).  Tests in `test_numbers.py` : `testNumericOperatorsModifierWithEscapes()`

**Review by oshadmi — APPROVED** [review]

**Pull request merged**